lib/mongo_storage.ex

```diff
@@ -228,4 +262,50 @@ defmodule ApplicationRunner.MongoStorage do
       :ok -> :ok
     end
   end
+
+  ####################
+  # Transaction      #
+  ####################
+
+  def start_transaction(env_id) do
+    with {:ok, session_pid} <- Mongo.Session.start_session(mongo_instance(env_id), :write),
+         session_uuid <- Ecto.UUID.generate(),
+         :yes <- Swarm.register_name(session_uuid, session_pid),
+         :ok <- Mongo.Session.start_transaction(session_pid) do
+      {:ok, session_uuid}
+    else
+      :no ->
+        BusinessError.error_during_transaction_start_tuple(%{
+          error: "uuid already used, please try again"
+        })
+
+      {:error, msg} ->
+        BusinessError.error_during_transaction_start_tuple(%{error_message: msg})
+    end
+  end
+
+  def commit_transaction(session_uuid, env_id) do
+    with :ok <- Mongo.Session.commit_transaction(Swarm.whereis_name(session_uuid)),
+         :ok <-
+           Mongo.Session.end_session(mongo_instance(env_id), Swarm.whereis_name(session_uuid)),
+         :ok <- Swarm.unregister_name(session_uuid) do
+      :ok
+    else
+      # TODO: check what to do in this case
+      _any ->
+        :error
+    end
+  end
+
+  def revert_transaction(session_uuid, env_id) do
+    with :ok <-
+           Mongo.Session.end_session(mongo_instance(env_id), Swarm.whereis_name(session_uuid)),
+         :ok <- Swarm.unregister_name(session_uuid) do
+      :ok
+    else
+      # TODO: check what to do in this case
```

Comment:
I'm not sure about this one. 
Does the end_session make sure that the data is not inserted into the db ? 
If that is the case we should handle the else like this:

if end_session failed, find why and log it, maybe the session crashed and in this case it is reverted by default. But if the session is still open, we must make sure that it is deleted.
If the unregister_name failed, what do we want to do ?